contracts/koh/test/ProposalsTest.js

```diff
@@ -0,0 +1,271 @@
+    let ballot_state = ballot.state;
+    assert.equal(ballot_state, 1, "Ballot box state is not PrepayingGas!");
+    let proposal = await koh.proposals(id)
+    let proposal_state = proposal.state;
+    assert.equal(proposal_state, 1, "Proposal state is not Started!");
+  });
+
+  it("cast vote invalid slot owner", async() => {
+    await koh.payForGas(id, 3);
+    await catchRevert(koh.castVote(id, 0, 1), "Invalid slot owner!");
+  });
+
+  it("cast vote invalid ballot box id", async() => {
+    await koh.setOwnerAllSlots();
+    await catchRevert(koh.castVote(200, 0, 1), "Invalid ballot box ID!");
+  });
+
+  it("cast vote before gas is paid", async() => {
+    await koh.setOwnerAllSlots();
+    await catchRevert(koh.castVote(id, 0, 1), "Ballot is not active!");
+  });
+
+  it("pay gas for more slots", async() => {
+    await catchRevert(koh.payForGas(id, 20), "Too many slots!");
+  });
+
+  it("vote change", async() => {
+    await koh.setOwnerAllSlots();
+    await koh.payForGas(id, 3);
+    // Test initial vote values
+    let vote = await koh.getVote(id, 0);
+    let pos_vote_count = await koh.getVoteCount(id, 1);
+    let neg_vote_count = await koh.getVoteCount(id, 2);
+    assert.equal(vote, 0, "Incorrect initial vote!");
+    assert.equal(pos_vote_count, 0, "Incorrect initial positive vote count!");
+    assert.equal(neg_vote_count, 0, "Incorrect initial negative vote count!");
+    // Cast positive vote
+    await koh.castVote(id, 0, 1);
+    vote = await koh.getVote(id, 0);
+    pos_vote_count = await koh.getVoteCount(id, 1);
+    assert.equal(vote, 1, "Incorrect vote!");
+    assert.equal(pos_vote_count, 1, "Incorrect positive vote count!");
+    // Change vote
+    await koh.castVote(id, 0, 2);
+    vote = await koh.getVote(id, 0);
+    assert.equal(vote, 2, "Incorrect vote change!");
+  });
+
+  // 1 negative 0 positive / -25% approval < rejection %
+  it("rejection during initial voting", async() => {
+    await koh.setOwnerAllSlots();
+    await koh.payForGas(id, 3);
+    // Check if states are correct
+    let ballot = await koh.ballot_boxes(id);
+    let ballot_state = ballot.state;
+    assert.equal(ballot_state, 2, "Ballot box state is not Active!");
+    let proposal = await koh.proposals(id)
+    let proposal_state = proposal.state;
+    assert.equal(proposal_state, 1, "Proposal state is not Started!");
+    // Cast negative vote
+    await koh.castVote(id, 0, 2);
+    await koh.updateProposalState(id);
+    // States shouldn't change during initial time
+    ballot = await koh.ballot_boxes(id);
+    ballot_state = ballot.state;
+    assert.equal(ballot_state, 2, "Ballot box state is not Active!");
+    proposal = await koh.proposals(id)
+    proposal_state = proposal.state;
+    assert.equal(proposal_state, 1, "Proposal state is not Started!");
+    await await web3.currentProvider.send(
+        {jsonrpc: "2.0", method: "evm_increaseTime", params: [proposal_start_period], id: 0}, (err, result) => {
```

Comment:
Can you add this to the utils.js as a function, so it's just one line of code here in the tests?